aldryn_people/models.py

```diff
@@ -276,9 +284,39 @@ def get_vcard(self, request=None):
         return vcard.serialize()
 
     def save(self, **kwargs):
+        language = self.get_current_language()
         if not self.slug:
-            unique_slugify(instance=self, value=self.name)
-        return super(Person, self).save(**kwargs)
+            self.slug = force_unicode(django_slugify(self.name))
+        # If there is still no slug, we must give it something to start with
+        if not self.slug:
+            self.slug = ugettext('unnamed-person')
+        if not Person.objects.language(language).filter(
+                translations__slug=self.slug).exclude(pk=self.pk).exists():
+            return super(Person, self).save(**kwargs)
+        for lang in LANGUAGE_CODES:
+            #
+            # We'd much rather just do something like:
+            # Person.objects.translated(lang, slug__startswith=self.slug)
+            # But sadly, this isn't supported by Parler/Django, see:
+            # http://django-parler.readthedocs.org/en/latest/api/\
+            #     parler.managers.html#the-translatablequeryset-class
+            #
+            slugs = []
+            all_slugs = (
+                Person.objects.language(lang)
```

Comment:
Yes, this is the subject of the next PR I've asked for a review of. Once that it merged into aldryn_translation_tools, the above code will be replaced. (Its working great locally already).